src/components/SearchBar/useSearch.js

```diff
@@ -5,15 +5,24 @@ export default function useSearch(
   cards,
   setNumPages,
   setCurrentPage,
-  setCards
+  setCards,
+  buildingType,
+  primaryCategory
 ) {
   const [criteria, setSearch] = useState({
     searchString: "",
     tags: {},
   });
 
   useEffect(() => {
```

Comment:
you can basically take out this whole useEffect! we are not concerned "about" a side effect right now. this was added because we wanted the search function to happen for every side effect which was adding a value to our search field. but now, we can literally have our onClick handler of our search button call the getCardsPagination